README.md

```diff
@@ -93,46 +95,66 @@ Migrations
+* `migrateToTagViews` scans over all your persistence ids and writes the tagged events to the `tag_views` table. At the same
+time it keeps track of its progress in the `tag_write_progress` table so if this were to fail due to say a C* issue you can resume
+and it won't start from scratch.
 
-* `delayed-event-timeout` has been replaces by `events-by-tag-gap-timeout` with the restriction removed that all
-events have to be tagged. 
+You do not need to worry if a small number of events are missed by `migrateToTagViews` are missed as they will be
+fixed during your `PersistentActor` recovery. However do not rely on this for full migration as only active `PersistentActor`s
+will be recovered and it will mean the start up time for your `PersistentActor`s will be very long.
 
+The following configuration changes. See `reference.conf` for full details:
+
+* `pubsub-minimum-interval` has been removed. If using DistributedPubsub notifications for tag writes then set `pubsub-notification` to on
+* `delayed-event-timeout` has been replaces by `events-by-tag.gap-timeout` with the restriction removed that all
+events have to be tagged. 
 * `first-time-bucket` format has changed to: `yyyyMMddTHH:mm` e.g. `20151120T00:00`
+* `eventual-consistency` has been removed. It may be re-added see [#263](https://github.com/akka/akka-persistence-cassandra/issues/263)
 
-* Drop the materialized view
+After you have migrated your data you can now remove the materialized view and `tagN` columns from the 
+`messages` table. It is *highly* recommended you do this as maintaining a materialized view is expensive
+and uses a large amount of capacity in your cluster:
 
+* Drop the materialized view
 ```cql
 DROP MATERIALIZED VIEW akka.eventsbytag;
 ```
-
 * Drop the tag columns on the messages table
-
 ```cql
 ALTER TABLE akka.messages DROP tag1;
 ALTER TABLE akka.messages DROP tag2;
 ALTER TABLE akka.messages DROP tag3;
-
 ```
 
-* Add tags column
+Note that the new `tags` column won't be back filled for old events. This won't affect 
+your `eventsByTag` queries as they come from a different table. This column is used for recovering any missed
+writes to the `tag_views` table when running persistent actors with the new version.
 
-```cql
-ALTER TABLE akka.messages ADD tags set<text>;
-```
+#### Gotchas
 
-Note that this won't be back ported for old events it will only be populated for new events. This won't affect 
-your `eventsByTag` queries as they come from a different table. This column is used for recovering any missed
-writes to the `tag_views` table.
+# Changing schema while your application is running
 
+Cassandra can generally handle schema changes to non-primary key columns. However all of the Journal's prepared statements
+that bind columns that you remove become invalid. So any migration step that involves dropping columns requires ActorSystems
+to be restarted.
```

Comment:
can these steps be done as a rolling upgrade or is a full cluster shutdown needed?
perhaps clarify that part

dropping columns and materialized view can be done afterwards, when all nodes are using the new version?

Reply:
actually going to remove this as there's no way a real user will hit this, I hit this as the test tries all the steps in the same JVM however this can't be for a real user as they need to upgrade their app before thinking about dropping columns, and once they have done that they won't have prepared statements for the columns they're about to drop

Reply:
ah, so `select *` is * at the time the stmt is executed?

Reply:
I allowed the journal to create all the tables so i didn't have to, then i dropped the messages table, put in the old one without the tags column and wrote to it manually. However the prepared statements in the journal were against the original table layout, not the one that is created manually in the test in the `beforeAll` 

So thinking about it it wasn't the dropping of the columns that caused this, it was the removing of loads of columns by dropping the table and adding a table with the same name with fewer columns